### PR TITLE
[VSM-187] Added drush command to import bulk redirects

### PR DESCRIPTION
**Pull request by FleetAdmiralButter**

### CODE REVIEW

File: tide_core.drush.inc [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+/**
+ * Implements hook_drush_command().
+ */
+function tide_core_drush_command() {
+  $items = [];
+
+  $items['redirect-import'] = [
+    'description' => "Imports a file and adds redirects to the redirect table",
+    'arguments' => [
+      'filepath' => 'Path to CSV (comma-separated), JSON or PHP file containing old and new URLs for redirects',
```
Comment (anthony-malkoun) [inline tide_core.drush.inc]: Can this be a URI?

File: tide_core.drush.inc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  ];
+
+  return $items;
+
+};
+
+/**
+ * Get the node id of the page to redirect to.
+ */
+function get_node_id($to) {
```
Comment (anthony-malkoun) [inline tide_core.drush.inc]: is this function getting a node id? It seems to be normalising URLs if anything if I understand it correctly.

File: tide_core.drush.inc [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+/**
+ * Uses redirect entity to save entry to DB.
+ */
+function add_redirects($from, $to) {
+  if ($from != '' && $to != '') {
+    $redirectrepo = redirect_repository();
+    if ($redirectrepo->findBySourcePath($from)) {
+      return;
+    }
+    $vicgov = $to == 'https://vic.gov.au' || $to == 'http://vic.gov.au' || $to == 'vic.gov.au';
```
Comment (anthony-malkoun) [inline tide_core.drush.inc]: This check is for just the homepage URLs? What happens if they come through with a trailing slash?

File: tide_core.drush.inc [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    $wwwvicgov = $to == 'https://www.vic.gov.au' || $to == 'http://www.vic.gov.au' || $to == 'www.vic.gov.au';
+    if ($vicgov || $wwwvicgov) {
+      $to = 'site-4/';
```
Comment (anthony-malkoun) [inline tide_core.drush.inc]: So `$to` is, I guess, a path prefix for certain URLs? Maybe that should just be a lookup in an array of redirect exceptions rather than the conditional statements above?

Comment (anthony-malkoun) [reply tide_core.drush.inc]: Ah no, looking at this, it is rewriting the whole `$to` value :+1:

File: tide_core.drush.inc [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    $queryStr = substr($from, 0, strpos($from, '?'));
+    if (!empty($queryStr)) {
+      $query = explode('&', $queryStr);
+    }
+    else {
+      $query = [];
+    }
+    $redirect->setSource($from, $query);
+    $redirect->setRedirect(get_node_id($to));
+    $redirect->getRedirect();
```
Comment (anthony-malkoun) [inline tide_core.drush.inc]: Why is this required?

**Comment by anthony-malkoun** [comment]
Hey @FleetAdmiralButter I was thinking about how to do this... would a migration be a good fit for this? Include `migration_csv` and I *think* redirects are supported as a destinations in a migration. Then could use drush to maintain them. This may be a completely bonkers idea :)

**Comment by FleetAdmiralButter** [comment]
> Just some random thoughts... These aren't things you need to change but more me getting an idea of how things are done around here :)
> 
> * Can we run `lint` locally? Just trying to get my head around why it took multiple commits to get it fixed so I suspect that we can't?

We should be able to run lint locally using `ahoy lint` but I haven't been able to get mine to run 😞, so that's why I ended up pushing the commits.

> * How do I test this? It would be good if some testing steps went into the description.

QA Steps would usually be in the description of the linked ticket on JIRA.

File: tide_core.drush.inc (hunk 1 above)
Comment (FleetAdmiralButter) [reply tide_core.drush.inc]: Supports both, full URL is used for redirecting from or to external sites but if either is internal the entry would just specify the internal URI link.

File: tide_core.drush.inc [hunk 6]
```diff
@@ -21,7 +35,302 @@ function tide_core_drush_command() {
...
+ *   Status code to return for redirect. Overridden if $to specifies code.
+ *
+ * @return bool
+ *   returns FALSE if input was invalid or a node id could not be found.
+ *
+ * @throws \Exception
+ *   If saving to the DB fails.
+ */
+function add_redirects($from, $to, $home_url, $lang, $status_code) {
+
```
Comment (alexdesignworks) [inline tide_core.drush.inc]: @anthony-malkoun how did this pass phpcs? Could you please check that this file is in a list of lint targets?
There should be no new line after `function`

**Comment by anthony-malkoun** [comment]
@FleetAdmiralButter what's happening with this PR?

File: tide_core.drush.inc (hunk 6 above)
Comment (FleetAdmiralButter) [reply tide_core.drush.inc]: This has been fixed @AlexSkrypnyk
